test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -739,4 +754,293 @@ TEST_VM_F(NMTVMATreeTest, SummaryAccountingWhenUseFlagInplace) {
   diff = tree.uncommit_mapping(0, 50, rd2);
   EXPECT_EQ(0, diff.tag[NMTUtil::tag_to_index(mtTest)].reserve);
   EXPECT_EQ(-50, diff.tag[NMTUtil::tag_to_index(mtTest)].commit);
-}
\ No newline at end of file
+}
+
+template<int NodeCount> struct ExpectedTree {
+  int nodes[NodeCount];
+  MemTag tags[NodeCount + 1];
+  VMATree::StateType states[NodeCount + 1];
+  NativeCallStackStorage::StackIndex res_si[NodeCount + 1];
+  NativeCallStackStorage::StackIndex com_si[NodeCount + 1];
+};
+template <int N>
+void check_tree(Tree& tree,const ExpectedTree<N>& et, int line_no) {
+  using Node = VMATree::TreapNode;
+  auto left_released = [&](Node n) -> bool {
+    return n.val().in.type() == VMATree::StateType::Released and
+           n.val().in.mem_tag() == mtNone;
+  };
+  auto right_released = [&](Node n) -> bool {
+    return n.val().out.type() == VMATree::StateType::Released and
+           n.val().out.mem_tag() == mtNone;
+  };
+  char for_this_node[50];
+  for (int i = 0; i < N; i++) {
+    VMATree::VMATreap::Range r = tree.tree().find_enclosing_range(et.nodes[i]);
+    ASSERT_TRUE(r.start != nullptr);
+    Node node = *r.start;
+    if (i == (N -1)) { // last node
+      EXPECT_TRUE(right_released(node)) << "right-of last node is not Released";
+      break;
+    }
+    if (i == 0) { // first node
+      EXPECT_TRUE(left_released(node)) << "left-of first node is not Released";
+    }
+
+    sprintf(for_this_node, "test at line: %d, for node: %d", line_no, et.nodes[i]);
```

Comment:
I see `warning: 'sprintf' is deprecated:...use snprintf(3) instead` on macOS when I try to run the gtest locally.

Reply:
Good catch. Forgotten to do so.
Changed to `snprintf` now.